test/test_torch.py

```diff
@@ -15055,6 +15055,32 @@ def test_cat(self, device, dtype):
         self.assertRaises(RuntimeError, lambda: torch.cat([]))
         self.assertRaisesRegex(TypeError, 'got None', lambda: torch.cat([x, None]))
 
+    def test_cat_different_dtypes(self, device):
```

Comment:
since what we're testing is the promotion behavior, this probably belongs in test_type_promotion.py.